### DISCUSSION TITLE
Text encoding percentage

### DISCUSSION THREAD

**Discussion by jysung** [TheLastBen/fast-stable-diffusion#321] [Q&A] [answered]
So a few days ago, was text encoding happening at 100% in the fast-dreambooth colab?
1. What exactly is text encoding? Any good documentation about what that is?
2. If I am training on a person with only 10 images, do you have any recommendations on what it should be set to? Thanks!

**Comment by TheLastBen** [comment, marked as answer]
Yes, before the update, the text_encoder was set to 100%

In simple terms, if it is set to 100% of the total training steps, you will get accurate representation at low steps count, but it will be harder to apply style to the trained subject, you will notice exaggerated overfitting.

If you want only photoreal images, you can set it to 100% and keep the steps lower (1500 steps for 10 input images).

If you want to stylize the trained subject, keep it around 40% and set the training steps count to 2500 for 10 input images.

**Comment by SAC020** [comment]
I am not clear either about what text encoding is. Does it require the input images to be named descriptively (such as "a woman holding a flower" or whatever description)? Where does it take the text from, how do you specify it?

**Reply by TheLastBen** [reply]
You don't need to add any text, just use a random identifier, and it will detect automatically the content of the instances pictures

**Reply by SAC020** [reply]
> You don't need to add any text, just use a random identifier, and it will detect automatically the content of the instances pictures

Actually I was hoping for an option to be able to specify the text myself, as I am trying to train the model for "content" which is less "vanilla" and not recognized normally by inverse image interrogations. 

Unless I am missing something, it's a vicious circle: the normal SD model does not recognize (very) NSFW terms, or deals with other terms very poorly (like limbs / feet), so I am trying to teach a model in order to bypass these limitations, but the training relies on the same limitations that triggered the need for training in the first place.

(hope I'm making sense)

**Reply by TheLastBen** [reply]
if you want to add text, rename the pictures to : word_word_word.jpg, separate the words by "_".
also you can use the old method, but you should know that adding text prompts will make the model non-retrainable, it will give bad results if retrained

**Reply by SAC020** [reply]
I do not plan to use the old method.

So, what I understand is:
- I can use the new method
- I need to specify word1_word2 in the picture name
- The model will need to be named the same (word1_word2)
- I cannot retrain the model even if I use the new method described above

Did I get it correctly?

**Reply by TheLastBen** [reply]
Yes, but you might be able to retrain in a limited way, depending on the prompt used

**Reply by SAC020** [reply]
Thank you

**Reply by TheLastBen** [reply]
happy to help

**Reply by SAC020** [reply]
Is it possible to have more than one subject in a model? If so, how should I name the files, and the model?

**Reply by TheLastBen** [reply]
Yes you can, you need to use a different name (identifier) for the images of the second subject and retrain the model
